### PR TITLE
amp-bind: Only get form service if needed

### LINKED ISSUE #8812: amp-bind shouldn't expect the ampform service to exist

### PR DESCRIPTION
**Pull request by choumx**
Fixes #8812.

- Fixes form error on load for pages without `amp-form`. 
- Bad suggestion from me in #8661. 😛 

/to @kmh287

### CODE REVIEW

File: extensions/amp-bind/0.1/bind-impl.js [hunk 1]
```diff
@@ -519,6 +498,28 @@ export class Bind {
...
 
   /**
+   * Observes the dynamic children of `element` for mutations, if any,
+   * for rescanning for bindable attributes.
+   * @param {!Element} element
+   */
+  observeElementIfNecessary_(element) {
+    const observeElement = elementToObserve => {
+      this.mutationObserver_.observe(elementToObserve, {childList: true});
+    };
```
Comment (jridgewell) [inline extensions/amp-bind/0.1/bind-impl.js]: Why isn't this a method on the instance?

Comment (kmh287) [inline extensions/amp-bind/0.1/bind-impl.js]: Could you clarify which method and instance you mean?

Comment (choumx) [inline extensions/amp-bind/0.1/bind-impl.js]: He means this arrow function can be an instance method. Made the change, though at a readability cost due to new ivar and method.

**Comment by choumx** [comment]
Woops, missed that. I'll close this since you started first.

File: extensions/amp-bind/0.1/bind-impl.js [hunk 2]
```diff
@@ -519,6 +502,35 @@ export class Bind {
...
 
   /**
+   * Observes the dynamic children of `element` for mutations, if any,
+   * for rescanning for bindable attributes.
+   * @param {!Element} element
+   * @private
+   */
+  observeDynamicChildrenOf_(element) {
+    if (typeof element.getDynamicElementContainers === 'function') {
+      element.getDynamicElementContainers().forEach(this.boundObserveElement_);
```
Comment (jridgewell) [inline extensions/amp-bind/0.1/bind-impl.js]: You could also have done `forEach(this.observeElement_, this)`, but this works too.

File: extensions/amp-bind/0.1/bind-impl.js [hunk 3]
```diff
@@ -519,6 +502,35 @@ export class Bind {
...
+    }
+  }
+
+  /**
+   * Observes `element` for child mutations.
+   * @param {!Element} element
+   * @private
+   */
+  observeElement_(element) {
+    this.mutationObserver_.observe(element, {childList: true});
```
Comment (jridgewell) [inline extensions/amp-bind/0.1/bind-impl.js]: Can we lazy init `#mutationObserver`?

Comment (choumx) [inline extensions/amp-bind/0.1/bind-impl.js]: Done.

File: extensions/amp-bind/0.1/bind-impl.js (hunk 2 above)
Comment (choumx) [inline extensions/amp-bind/0.1/bind-impl.js]: I like that better, done.

**Pull request merged**